### PR TITLE
Bug 1866693: docs: enabling RHCOS extensions using MachineConfig

### PR DESCRIPTION
**Pull request by sinnykumari**

### CODE REVIEW

**Comment by sinnykumari** [comment]
corresponding doc pr to enable extensions during cluster install - https://github.com/openshift/installer/pull/4018

**Comment by ashcrow** [comment]
Ah wait the item that looks like code change was docs too! :facepalm: :laughing:

File: docs/MachineConfiguration.md [hunk 1]
```diff
@@ -180,6 +181,29 @@ spec:
 
 **Note:** The RT kernel lowers throughput (performance) in return for improved worst-case latency bounds. This feature is intended only for use cases that require consistent low latency. For more information, see the [Linux Foundation wiki](https://wiki.linuxfoundation.org/realtime/start) and the [RHEL RT portal](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_real_time/8/).
 
+### RHCOS Extensions
+RHCOS is a minimal OS which provides capabilities common across all the platforms. With extensions support, OCP 4.6 and onward users can enable additional functionalities on the RHCOS nodes. Currently(OCP 4.6) supported extensions are `usbguard` and `kernel-devel`.
```
Comment (ashcrow) [inline docs/MachineConfiguration.md:185]:
```suggestion
RHCOS is a minimal OCP focused OS which provides capabilities common across all the platforms. With extensions support, OCP 4.6 and onward users can enable a limited set of additional functionality on the RHCOS nodes. In OCP 4.6 the supported extensions are `usbguard` and `kernel-devel`.
```

Comment (sinnykumari) [reply docs/MachineConfiguration.md]: updated with suggestion.

**Comment by kikisdeliveryservice** [comment]
❤️ docs. This is great.

/approve

**Comment by sinnykumari** [comment]
needs lgtm blessing :)

File: docs/MachineConfiguration.md [hunk 2]
```diff
@@ -180,6 +181,29 @@ spec:
 
 **Note:** The RT kernel lowers throughput (performance) in return for improved worst-case latency bounds. This feature is intended only for use cases that require consistent low latency. For more information, see the [Linux Foundation wiki](https://wiki.linuxfoundation.org/realtime/start) and the [RHEL RT portal](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_real_time/8/).
 
+### RHCOS Extensions
+RHCOS is a minimal OCP focused OS which provides capabilities common across all the platforms. With extensions support, OCP 4.6 and onward users can enable a limited set of additional functionality on the RHCOS nodes. In OCP 4.6 the supported extensions are `usbguard` and `kernel-devel`.
+
+**Note:** `kernel-devel` extension should be installed only when you are using compatible traditional kernel (kernleType: default). If a user tries to install on an incompatible kernel (e.g. kernelType: realtime), MachineConfig will not be rendered and relevant errors can be seen in the machine-config-controller pod.
```
Comment (cgwalters) [inline docs/MachineConfiguration.md:187]: Let's do docs PRs in a hackmd.io or something to start?  There's e.g. a `kernleType` typo and other things that would be simpler to just fix and not do a review/repush cycle.

**Comment by cgwalters** [comment]
/lgtm
to get this in, we can do more as followup.

**Comment by ashcrow** [comment]
/test images

(doesn't look like prow honors doc tags)

**Pull request merged**